week1/assignment3-2/ex4.cpp

```diff
@@ -0,0 +1,244 @@
+                                 "+358 40 123 4567", {}};
+
+    // Create Account-struct
+    Account account = {300};
+
+    // Connect customer to account (account_number serves also as 
+    // a map key)
+    example_customer.accounts[account_number] = account;
+
+    // Connect customer to a bank (customer_number serves also as 
+    // a map key)
+    bank[customer_number] = example_customer;
+
+    // User interface
+    std::cout << "Welcome to the Elisa's Bank!\n"; 
+    std::cout << "Please choose an action:\n";
+    bool stay_in_loop = false;
+
+    // Ask a user to choose an action (in the while loop) until 
+    // the user choose to quit the program
+    while (true) {
+        std::cout << "\n\n1: select a customer\n";
+        std::cout << "2: create a new customer\n";
+        std::cout << "3: exit\n";
+        int selection{};
+        std::cin >> selection;
+
+        switch (selection) {
+            case 1:
+                {
+                stay_in_loop = true;  // Needed for exiting from next while-loop
+                std::cout << "Enter the customer number: ";
+                int c_num{};
+                std::cin >> c_num;
+                //current_customer = bank[c_num];
+                std::cout << "\n\nCustomer '" << bank[c_num].name << "' chosen\n";
+
+                while(stay_in_loop) {
+
+                    std::cout << "\n\nPlease, choose an action:\n";
+                    std::cout << "1: check balance\n";
+                    std::cout << "2: deposit money\n";
+                    std::cout << "3: withdraw money\n";
+                    std::cout << "4: create a new account\n";
+                    std::cout << "5: exit to previous menu\n\n";
+
+                    int choice{};
+                    std::cin >> choice;
+
+                    switch (choice)
+                    {
+                    case 1:  
+                        {
+                        int chosen_acc_num = get_account_number();
+                        std::cout << "\n\nAccount balance is " 
+                        // From bank-map, find right customer with the customer number
+                        // and then find right account with the account number and pick
+                        // it's balance
+                            << bank[c_num].accounts[chosen_acc_num].balance << " €\n";
+                        }
+                        break;
+                    case 2:  
+                        {
+                        int chosen_acc_num = get_account_number();
+                        bank[c_num] = deposit_money(bank[c_num], chosen_acc_num);
+                        }
+                        break;
+                    case 3:  
+                        {
+                        int chosen_acc_num = get_account_number();
+                        bank[c_num] = withdraw_money(bank[c_num], chosen_acc_num);
+                        }
+                        break;
```

Comment:
These work as a way to create new accounts, since there's no check if that account is taken or not. It can also create same account number that some other customer may have. But if user inputs are correct and they don't try to break it, it works! 😄